requests/utils.py

```diff
@@ -167,7 +167,7 @@ def get_netrc_auth(url, raise_errors=False):
 
         for f in NETRC_FILES:
             try:
-                loc = os.path.expanduser('~/{0}'.format(f))
+                loc = os.path.expanduser(f)
```

Comment:
So to make this more test-able and to probably make this more future-proof, what if we do the following:

Let's revert the change to L37, so that it looks like:

```py
NETRC_FILES = ('.netrc', '_netrc')
```

Then let's do this on L162 or so:

```py
netrc_file = os.environ.get('NETRC')
if netrc_file is not None:
   netrc_locations = (netrc_file,)
else:
   netrc_locations = ('~/{0}'.format(f) for f in NETRC_FILES)
```

And then we can change the loop on L168 to be:

```py
for f in netrc_locations:
```

Sure it's slightly more code, but it also is far more explicit and gives us room to expand if necessary.